packages/lab/src/__tests__/__e2e__/skip-link/SkipLink.cy.tsx

```diff
@@ -0,0 +1,109 @@
+import { SkipLink } from "../../../skip-link";
+
+const composedStories = composeStories(skipLinkStories);
+const { Default, MultipleLinks } = composedStories;
+
+export const NoTargetRef = () => {
+  return (
+    <>
+      <div>
+        <span style={{ height: 50, lineHeight: "50px" }} tabIndex={-1}>
+          Click here and press the Tab key to see the Skip Link
+        </span>
+        <div style={{ position: "relative", maxWidth: 500 }}>
+          <SkipLink data-testid="skipLink" href="#main">
+            Skip to main content
+          </SkipLink>
+
+          <div
+            style={{
+              borderTop: "2px solid grey",
+              fontSize: 24,
+              lineHeight: 3.5,
+            }}
+          >
+            What we do
+          </div>
+
+          <article id="main">
+            <section>
+              <h1>UI Toolkit</h1>
+              <p>
+                UITK provides you with a suite of UI components and a flexible
+                theming system. With no customisation, the default theme offers
+                an attractive and modern look-and-feel, with both light and dark
+                variants and support for a range of UI densities. We have
+                included a theming system which allows you to easily create
+                theme variations, or in fact substitute alternate themes.
+              </p>
+            </section>
+            <section>
+              <h1>Goals</h1>
+              <p>
+                The UITK has been developed with the following design goals:
+              </p>
+              <ul>
+                <li>
+                  Providing a comprehensive set of commonly-used UI controls
+                </li>
+                <li>Complying with WCAG 2.1 accessibility guidelines</li>
+                <li> To be lightweight and performant</li>
+                <li> Offering flexible styling and theming support</li>
+                <li> Minimizing dependencies on third-party libraries</li>
+              </ul>
+            </section>
+          </article>
+        </div>
+      </div>
+    </>
+  );
+};
+
+describe("GIVEN a SkipLink", () => {
+  checkAccessibility(composedStories);
+  describe("WHEN there is a single SkipLink", () => {
+    it("THEN it should move focus to the target element when interacted with", () => {
+      cy.mount(<Default />);
+      cy.findByText(
+        "Click here and press the Tab key to see the Skip Link"
+      ).click();
+      cy.realPress("Tab");
+      let skipLink = cy.findByTestId("skipLink");
+      skipLink.should("be.visible");
+      skipLink.click();
+
+      cy.get("#main").should("be.focused");
+    });
+
+    it("THEN it should not move focus if no target ref is given", () => {
+      cy.mount(<NoTargetRef />);
+      cy.findByText(
+        "Click here and press the Tab key to see the Skip Link"
+      ).click();
+      cy.realPress("Tab");
+      let skipLink = cy.findByTestId("skipLink");
```

Comment:
I'm not a big fan of storing references in tests like this. I'd much rather the repetition of calling cy.findByTestId("skipLink")